src/transformers/integrations/integration_utils.py

```diff
@@ -221,14 +225,18 @@ def _objective(trial, checkpoint_dir=None):
                 if key != "local_rank":
                     setattr(trainer.args, key, value)
             trainer.train(resume_from_checkpoint=None)
+            # If we want to load the best model, override the object with best
+            # metric objective.
+            if use_best_model:
+                trainer.objective = trainer.compute_objective(trainer.state.best_metrics)
```

Comment:
This block, a few line below, set `trainer.objective`, and at the end it returns `None`. Is this why the above block needs to access `trainer.objective`?

Reply:
(Have to say I don't know well about the reason of this `else` block exist)

Reply:
I really don't know what the else block here is either. Seems like it's related to distributed training? But I was struggling to figure out how this is used, since it doesn't return anything.

Here are the PRs that added it:
https://github.com/huggingface/transformers/pull/18931
https://github.com/huggingface/transformers/pull/19002